HMNFD_Design.md

```diff
@@ -0,0 +1,274 @@
+# Cray State Change Notification Service (hmnfd)
+
+The Shasta State Change Notification (SCN) fanout service provides the ability
+to notify subscribers of component hardware state changes and other changes
+made to and by the Hardware State Manager.  _hmnfd_ is a companion service
+to the Hardware State Manager in that it does the heavy lifting of 
+distributing state changes and managing subscriptions.
+
+This service uses a RESTful interface to provide the following functions:
+
+* Subscription for component state changes:
+  * Hardware state (On, Off, Ready, etc.)
+  * Logical state (arbitrary, e.g., AdminDown)
+  * Role (Compute, NCN, etc.)
+  * Enabled state
+  * Flag (OK, Alert, Warning, etc.)
+* Ability to view current subscriptions
+* Ability to delete subscriptions
+* Ability to get and modify current service operating parameters
+* Ability to receive State Change Notifications from the Hardware State Manager
+
+_hmnfd_ typically runs on an SMS cluster as one or more Docker container 
```

Comment:
I dont think SMS is a thing anymore.   I think I would say: kubernetes cluster in the services mesh

Reply:
Fixed.